File: benchmarks/projects/todo-js/src/taskStore.js
export class TaskStore {
  #tasks = [];
  #nextId = 1;

  create(title) {
    const task = {
      id: `task-${this.#nextId++}`,
      title,
      completed: false
    };
    this.#tasks.push(task);
    return { ...task };
  }

  update(taskId, updater) {
    const index = this.#tasks.findIndex((task) => task.id === taskId);
    if (index === -1) {
      throw new Error(`Task not found: ${taskId}`);
    }
    const updated = updater(this.#tasks[index]);
    this.#tasks[index] = updated;
    return { ...updated };
  }

  list() {
    return this.#tasks.map((task) => ({ ...task }));
  }
}

File: benchmarks/projects/todo-python/src/task_store.py
from copy import deepcopy


class TaskStore:
    def __init__(self) -> None:
        self._tasks = []
        self._next_id = 1

    def create(self, title: str) -> dict:
        task = {
            "id": f"task-{self._next_id}",
            "title": title,
            "completed": False,
        }
        self._next_id += 1
        self._tasks.append(task)
        return deepcopy(task)

    def update(self, task_id: str, updater) -> dict:
        for index, task in enumerate(self._tasks):
            if task["id"] == task_id:
                updated = updater(deepcopy(task))
                self._tasks[index] = deepcopy(updated)
                return deepcopy(updated)
        raise ValueError(f"Task not found: {task_id}")

    def list(self) -> list:
        return deepcopy(self._tasks)
